### ISSUE TITLE
url_title not detecting some special characteres

### ISSUE THREAD

**Issue by mmacagnan** [codeigniter4/CodeIgniter4#3038]
Hey, it's so nice to participate in an improvement for this spectacular framework.
Yesterday I commit a change to a client system. When he try to create a resource all seems to be fine, but i used the url_title to slugify the title to url and save it to the database.

The problem is that did not detected some special characteres like ç,~,`

![image](https://user-images.githubusercontent.com/40900911/83026752-c2b2cd80-a005-11ea-8742-b4d694a449dd.png)

![image](https://user-images.githubusercontent.com/40900911/83026880-cc3c3580-a005-11ea-901c-4b7a35da2a6d.png)

**Comment by natanfelles** [comment]
Try like this

```php
function url_segment(string $text) : string
{
	$text = url_title($text, '-', true);
	helper('text');
	$text = convert_accented_characters($text);
	return $text;
}
```

**Comment by lonnieezell** [comment]
@natanfelles is absolutely correct. For a deeper explanation the HTTP spec ([RFC-3986](https://www.ietf.org/rfc/rfc3986.txt)) does not allow characters outside of the [US-ASCII](http://www.columbia.edu/kermit/ascii.html) as part of a URI, unless it's been encoded. 

That being said, I'm wondering if we should incorporate that into the `url_title()` helper method itself, or just make a note of it in the docs. I'm leaning toward docs because otherwise it might be a surprise to developers.

@natanfelles @MGatner @michalsn what do you guys think?

**Comment by natanfelles** [comment]
The `convert_accented_characters` should be incorporated.

**Comment by mmacagnan** [comment]
Thank you.
It would be nice if the functionality of convert_accented_characters is implemented in url_title

**Comment by michalsn** [comment]
I remember many discussions about this one in the past. Personally, I would love to see `url_title()` handling accented characters under the hood, but at this point, it's too late for that since developers may rely on produced results heavily.

Another option would be to introduce yet another parameter to `url_title()` function, but idk if I would like to see this. 

If I gonna need this functionality I will probably write a simple helper for that, just like I always did in CI3.

**Comment by lonnieezell** [comment]
I think a good solution would be to create a new helper function in the url_helper: `mb_url_title`. 
Thoughts? Anyone want to submit a PR?

**Comment by michalsn** [comment]
Resolved via #3086
